root/programs/C#/Frameworks/Infrastructure/Public/Db/DamManagedOdp/DamManagedOdp.cs

```diff
@@ -0,0 +1,908 @@
+        /// 通常、Ｄａｏ経由で利用する。
+        /// </remarks>
+        public override void SetUserParameter(string userParameterName, string userParameterValue)
+        {
+            if (this.IsDPQ)
+            {
+                // 動的パラメタライズド クエリ
+
+                // パラメタを保存（実行時に解析してまとめて設定する）
+                this._userParameter.Add(userParameterName, userParameterValue);    
+            }
+            else
+            {
+                // 通常のパラメタライズド クエリ
+
+                // パラメタを保存（操作ログで使用する用途）
+                this._userParameter.Add(userParameterName, userParameterValue);
+
+                // 現在のコマンドテキストを取得
+                string commandText = this._cmd.CommandText;
+
+                // ユーザパラメタを指定の文字列で置換する
+                commandText = commandText.Replace("%" + userParameterName + "%", userParameterValue);
+
+                // 置換後のコマンドテキストを設定
+                this._cmd.CommandText = commandText;
+            }
+        }
+
+        #endregion
+
+        #region SQLの実行
+
+        #region SQL実行前の制御用メソッド
+
+        /// <summary>
+        /// SQL実行前の、
+        /// ・通常のパラメタライズド クエリ
+        /// ・動的パラメタライズド クエリ
+        /// 制御用メソッド
+        /// </summary>
+        private void PreExecQuery()
+        {
+            if (this.IsDPQ)
+            {
+                // 動的パラメタライズド クエリ
+
+                // パラメタライズド クエリに変換する（各タグの処理、IN句の処理）。
+                this._cmd.CommandText = this.Convert(DamManagedOdp._paramSign);
+
+                // パラメタを設定する前に、this._QueryStatusをSQL
+                // （通常のパラメタライズド・クエリ）に設定する。
+                // ※ SetParameter内での動作が変わるため。
+                this._QueryStatus = DbEnum.QueryStatusEnum.SPQ;
+
+                #region Dictionaryのパラメタを設定
+
+                foreach (string paramName in this._parameter.Keys)
+                {
+
+                    //#region メタル対応
+
+                    //// 存在しないパラメタを設定しようとしている場合
+                    //if (this._cmd.CommandText.IndexOf(DamManagedOdp._paramSign + paramName) == -1)
+                    //{
+                    //    continue; // スキップする。
+                    //}
+
+                    //#endregion
+
+                    // 2008/10/24---null・DBNull対応（下記nullチェックを追加）
+                    // ここはforeachで取るので「キーなし」にならない
+                    if (this._parameter[paramName] == null)
+                    {
+                        // パラメタがnullの場合
+                        // ※ 下記のtypeof(ArrayList).ToString()対策
+
+                        // 既定値対策されているのでこのままで良い。
+                        this.SetParameter(
+                            paramName,
+                            this._parameter[paramName],
+                            this._parameterType[paramName],
+                            (int)this._parameterSize[paramName],
+                            (ParameterDirection)this._parameterDirection[paramName]);
+                    }
+                    else if (this._parameter[paramName].GetType() == typeof(ArrayList))
```

Comment:
@daisukenishino 

The developers can use `ArrayList` object only.
To enable to use other types, like `string[]` or `List<T>`, it might be a good idea to change `ArrayList` to `IList`.

Reply:
@daisukenishino 

[DamOraOdp](https://github.com/OpenTouryoProject/OpenTouryo/blob/develop/root/programs/C%23/Frameworks/Infrastructure/Public/Db/DamOraOdp/DamOraOdp.cs#L657) class is not revised. (Other dam classes are all revised.)